### PR TITLE
Add support for inline assembly

### PR DESCRIPTION
**Pull request by Jezurko**
TODO: 
- [x] Add simple asm statement builder
- [x] Cleanup printing with empty ins/outs
- [x] Tests
- [x] description in TableGen
- [ ] figure out if the `inline` qualifier is relevant and if it is accessible
- [x] (?) merge into HighLevelOps.td

### CODE REVIEW

File: include/vast/CodeGen/CodeGenStmtVisitor.hpp [hunk 1]
```diff
@@ -407,8 +408,99 @@ namespace vast::cg {
...
+            auto get_out_expr       = [&](int i) { return stmt->getOutputExpr(i); };
+            auto get_out_name       = [&](int i) { return stmt->getOutputName(i); };
+            auto get_out_constraint = [&](int i) { return stmt->getOutputConstraint(i); };
+
+            auto get_in_expr       = [&](int i) { return stmt->getInputExpr(i); };
+            auto get_in_name       = [&](int i) { return stmt->getInputName(i); };
+            auto get_in_constraint = [&](int i) { return stmt->getInputConstraint(i); };
+
+            int arg_num = 0;
+            auto fill_vectors = [&](int size, auto &get_expr, auto &get_name,
```
Comment (xlauko) [inline include/vast/CodeGen/CodeGenStmtVisitor.hpp:450]: most of these can be `const auto &`

File: include/vast/CodeGen/CodeGenStmtVisitor.hpp [hunk 2]
```diff
@@ -407,8 +408,99 @@ namespace vast::cg {
         // Assembly Statements
         //
 
-        // operation VisitAsmStmt(const clang::AsmStmt *stmt)
-        // operation VisitGCCAsmStmt(const clang::GCCAsmStmt *stmt)
+        // operation VisitAsmStmt(const clang::AsmStmt *stmt);
+        operation VisitGCCAsmStmt(const clang::GCCAsmStmt *stmt) {
+            auto &ctx = context().mctx;
```
Comment (xlauko) [inline include/vast/CodeGen/CodeGenStmtVisitor.hpp:413]: we have `mcontext()`

File: test/vast/Dialect/HighLevel/asm-b.c [hunk 3]
```diff
@@ -0,0 +1,12 @@
+// RUN: %vast-front -vast-emit-mlir=hl %s -o - | %file-check %s
+// RUN: %vast-front -vast-emit-mlir=hl %s -o %t && %vast-opt %t | diff -B %t -
+
+void f() {
+int src = 1;
+int dst;
+// CHECK: hl.inline_asm {is_volatile} #core.strlit<"mov %1, %0\n\tadd $1, %0">([0] %2 : [#core.strlit<"=r">]) (ins : [1] %4 : [#core.strlit<"r">]) () () : (!hl.lvalue<!hl.int>, !hl.int) -> ()
```
Comment (xlauko) [inline test/vast/Dialect/HighLevel/asm-b.c:7]: can you add #core.strlit alias to make this less verbose?

File: include/vast/Dialect/HighLevel/HighLevelOps.td [hunk 4]
```diff
@@ -1173,4 +1174,46 @@ def HighLevel_EmptyDeclOp
   let assemblyFormat = [{ attr-dict }];
 }
 
+def InlineAsmOp
+  : HighLevel_Op< "inline_asm", [AttrSizedOperandSegments] >
```
Comment (xlauko) [inline include/vast/Dialect/HighLevel/HighLevelOps.td:1178]: shorten the name to just asm?

**Review by xlauko — APPROVED** [review]

File: test/vast/Dialect/HighLevel/asm-b.c (hunk 3 above)
Comment (Jezurko) [reply test/vast/Dialect/HighLevel/asm-b.c]: I have tried and struggled with outputing "special" symbols via `llvm::raw_ostream`. They are show as their ascii codes instead.

**Review by xlauko — APPROVED** [review]

**Pull request merged**